server/src/main/java/org/opensearch/autotagging/RuleValidator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.autotagging;
+
+import org.opensearch.common.ValidationException;
+import org.joda.time.Instant;
+
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.cluster.metadata.QueryGroup.isValid;
+
+/**
+ * This is the validator for rule. It ensures that the rule has a valid description, label,
+ * update time, attribute map, and the rule adheres to the feature type's constraints.
+ *
+ * @opensearch.experimental
+ */
+public class RuleValidator<T extends FeatureType> {
+    private final String description;
+    private final Map<Attribute, Set<String>> attributeMap;
+    private final String label;
+    private final String updatedAt;
+    private final T featureType;
+    private final ValidationException validationException = new ValidationException();
+
+    public RuleValidator(String description, Map<Attribute, Set<String>> attributeMap, String label, String updatedAt, T featureType) {
+        this.description = description;
+        this.attributeMap = attributeMap;
+        this.label = label;
+        this.updatedAt = updatedAt;
+        this.featureType = featureType;
+    }
+
+    public void validate() {
+        requireNonNullOrEmpty(description, "Rule description can't be null or empty");
+        requireNonNullOrEmpty(label, "Rule label can't be null or empty");
+        requireNonNullOrEmpty(updatedAt, "Rule update time can't be null or empty");
+
+        if (featureType == null) {
+            validationException.addValidationError("Couldn't identify which feature the rule belongs to. Rule feature can't be null.");
+        }
+        if (attributeMap == null || attributeMap.isEmpty()) {
+            validationException.addValidationError("Rule should have at least 1 attribute requirement");
+        }
+        if (updatedAt != null && !isValid(Instant.parse(updatedAt).getMillis())) {
+            validationException.addValidationError("Rule update time is not a valid epoch");
+        }
+        if (attributeMap != null && featureType != null) {
+            validateAttributeMap();
+        }
+        if (!validationException.validationErrors().isEmpty()) {
+            throw new IllegalArgumentException(validationException);
+        }
+    }
+
+    public void validateAttributeMap() {
+        for (Map.Entry<Attribute, Set<String>> entry : attributeMap.entrySet()) {
+            Attribute attribute = entry.getKey();
+            Set<String> attributeValues = entry.getValue();
+            if (featureType.getAttributeFromName(attribute.getName()) == null) {
+                validationException.addValidationError(
+                    attribute.getName() + " is not a valid attribute within the " + featureType.getName() + " feature."
+                );
+            }
```

Comment:
Can we refactor these validation blocks into methods to make this function look more readable ?